index.js

```diff
@@ -1,29 +1,62 @@
-/* globals require, __dirname, process */
-
 const path = require('path');
+const builder = require('adguard-hostlists-builder');
 const fs = require('fs');
-const builder = require("adguard-hostlists-builder");
-
-// Validate services.json and make sure it is a valid JSON.
-try {
-    JSON.parse(fs.readFileSync('assets/services.json', 'utf8'));
-} catch (ex) {
-    console.error('Failed to parse services.json', ex);
-    process.exit(1);
-}
-
-// Compile hostlists.
+const { checkRemovedServices } = require('./scripts/services/check-removed-services');
+const { rewriteServicesJSON } = require('./scripts/services/rewrite-services-json');
 
 const filtersDir = path.join(__dirname, './filters');
 const assetsDir = path.join(__dirname, './assets');
 const tagsDir = path.join(__dirname, './tags');
 const localesDir = path.join(__dirname, './locales');
+const inputServicesDir = path.join(__dirname, './services');
+const outputServicesFile = path.join(assetsDir, 'services.json');
 
+/**
+ * Validate services.json and make sure it is a valid JSON.
+ *
+ * @param {string} filePath The file path for the "services.json" file.
+ * @throws {Error} If JSON file is not valid.
+ */
+const validateJson = (filePath) => {
+    try {
+        JSON.parse(fs.readFileSync(filePath, 'utf8'));
+    } catch (error) {
+        console.error('Failed to parse services.json', error);
+        process.exit(1);
+    }
+};
+
+/**
+ * Builds the "services.json" file by performing the following steps:
```

Comment:
`"services.json"` is not hardcoded into the function so it should not be mentioned in the description like this. param name should be used instead of it:
```suggestion
 * Builds the result services file and saves it to `resultFilePath`.
 * During the build the following steps are performed:
 ```